controlpanel/core/auth/oidc.py

```diff
@@ -0,0 +1,128 @@
+from django.conf import settings
+from django.dispatch import receiver
+from django.urls import reverse
+from django.utils import timezone
+
+import structlog
+from authlib.integrations.django_client import OAuth, OAuthError, token_update
+
+from controlpanel.core.models.user import User
+
+logger = structlog.get_logger(__name__)
+
+oauth = OAuth()
+oauth.register("auth0", **(settings.AUTHLIB_OAUTH_CLIENTS["auth0"]))
+
+
+@receiver(token_update)
+def on_token_update(sender, token, refresh_token=None, access_token=None, **kwargs):
+    """for information purpose only
+    Since this app (so far) doesn't use access_token linking to the user directly
+    the refresh_token is not very useful here, even app will use the access_token
+    in the future,  using the auto token_update signal from authlib, the problem will
+    be where we store the tokens (access_token and refresh_token), if we store them in
+    the session, this token_update won't allow app to access current session object
+    from current request, alternatively we can use django-crequest or refresh the
+    access_token manually
+    """
+    if refresh_token:
+        logger.info("The token has been refreshed by using refresh_token")
+
+
+class OIDCSubAuthenticationBackend:
+    def __init__(self, token):
+        self.token = token
+
+    def filter_users_by_claims(self):
+        user_id = self.token.get("userinfo", {}).get("sub")
+        if not user_id:
+            return User.objects.none()
+
+        try:
+            user = User.objects.get(pk=user_id)
+            return user
+        except User.DoesNotExist:
+            return User.objects.none()
```

Comment:
Perhaps this is personal preference, but you could reduce number of lines by using:
```suggestion
        return User.objects.filter(pk=user_id).first()
```
Will return the user if found, otherwise returns `None` even if the `user_id` is `None`